cpp/src/interop/to_arrow.cu

```diff
@@ -398,9 +398,9 @@ std::shared_ptr<arrow::Table> to_arrow(table_view input,
     arrays.end(),
     metadata.begin(),
     std::back_inserter(fields),
-    [](auto const& array, auto const& meta) { return arrow::field(meta.name, array->type()); });
+    [](auto const& array, auto const& meta) { return std::make_shared<arrow::Field>(meta.name, array->type()); });
```

Comment:
@trxcllnt should we upstream this issue to the arrow devs indicating that the library contained in the wheels isn't exporting all the symbols that we would expect?

Reply:
I made this change before we realized the ABI was different... maybe the symbol really is there? Can we revert this and see if it still breaks?

Reply:
Hmm I thought that I did that, but maybe that was also before I was correctly forwarding along the ABI flags from the Python to the C++ build here. Let me test again to be sure.

Reply:
Yup I guess I was wrong. This is in fact not necessary :tada: